Tests/SonarScanner.MSBuild.TFS.Tests/BuildVNextCoverageSearchFallbackTests.cs

```diff
@@ -74,6 +77,17 @@ public void Fallback_FilesLocatedCorrectly()
             TestUtils.CreateTextFile(dir, "Duplicate.coverage", ""); // appears in both places - only one should be returned
             var expected3 = TestUtils.CreateTextFile(subDir, "BAR.COVERAGE", ""); // should be found
 
+            filesToDelete.AddRange(new List<string>()
+            {
+                Path.Combine(dir, "foo.coverageXXX"),
+                Path.Combine(dir, "abc.trx"),
+                Path.Combine(dir, "foo.coverage"),
+                Path.Combine(dir, "DUPLICATE.coverage"),
+                Path.Combine(dir, "BAR.coverage.XXX"),
+                Path.Combine(dir, "Duplicate.coverage"),
+                Path.Combine(subDir, "BAR.COVERAGE")
+            });
```

Comment:
There quite a bit of duplication in the file tracking, and sometimes the creation of a file is separate from adding it to the list -> potential for human error when maintaining the tests in future.

I was trying to think how to consolidate the code, given that the tests are in separate classes and there is state that needs to be tracked. The best I could come up with was a static class: something like:
```
// Tracks test files that need to be deleted in the Azure pipeline
// so they don't interfere with the Azure tests
internal static class FileDeletionTracker
{
   private static List<string> filesToDelete;
   public static void CreateTextFile(...) {} // create file and add to list
   puvoid static void SaveFileList() { } // save to env var to be picked up by in Azure pipeline. Called from [ClassCleanup]
  
}
```

What do you think?

Reply:
So all of this code has been removed, and the files created in concerned unit tests are now created in %TEMP% folder instean of the temp directory of the build agent to avoid them to be picked up. 
However, i still needed to let at least this bit in the powershell that removed the files previously : Get-ChildItem $(Agent.TempDirectory) -Filter 'VSCodeCoverageReport.*' -Recurse -Attributes !Directory | Remove-Item

because i don't where this file created comes from, still in the agent temp directory (i hadn't check all the tests though)